blocks/identity-block/components/SocialSignOn/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+import PropTypes from '@arc-fusion/prop-types';
+import useIdentity from '../Identity';
+import './styles.scss';
+
+const SocialSignOn = ({ onError, redirectURL }) => {
+  const { Identity } = useIdentity();
+  const [config, setConfig] = useState(() => Identity?.configOptions ?? {});
+  const [isGoogleInitialized, setIsGoogleInitialized] = useState(false);
+  const [isFacebookInitialized, setIsFacebookInitialized] = useState(false);
+
+  const isGoogleReset = useRef(false);
+
+  if (!isServerSide() && !window.onFacebookSignOn) {
+    window.onFacebookSignOn = async () => {
+      try {
+        await Identity.facebookSignOn();
+        window.location = redirectURL;
+      } catch (e) {
+        onError();
+      }
+    };
+  }
+
+  useEffect(() => {
+    const fetchConfig = async () => {
+      await Identity.getConfig();
+      setConfig(Identity.configOptions);
+    };
+    if (Identity && !Identity.configOptions) {
+      fetchConfig();
+    }
+  }, [Identity]);
+
+  useEffect(() => {
+    const initializeFacebook = async () => {
+      await Identity.initFacebookLogin(null);
+    };
+    const initializeGoogle = async () => {
+      await Identity.initGoogleLogin(null, {
+        width: 300,
+        height: 48,
+        onSuccess: () => {
+          if (!isGoogleReset.current) {
+            isGoogleReset.current = true;
+          } else {
+            window.location = redirectURL;
+          }
+        },
+        onFailure: () => { onError(); },
+      });
+    };
+    if (config.googleClientId && !isGoogleInitialized) {
+      initializeGoogle();
+      setIsGoogleInitialized(true);
+    }
+    if (config.facebookAppId && !isFacebookInitialized) {
+      initializeFacebook();
+      setIsFacebookInitialized(true);
+    }
+  }, [Identity,
+    config.facebookAppId,
+    config.googleClientId,
+    isFacebookInitialized,
+    isGoogleInitialized,
+    onError,
+    redirectURL]);
```

Comment:
We should split this up so that the effects for each do not cause a re-check/init for the other...
```suggestion
  useEffect(() => {
    const initializeGoogle = async () => {
      await Identity.initGoogleLogin(null, {
        width: 300,
        height: 48,
        onSuccess: () => {
          if (!isGoogleReset.current) {
            isGoogleReset.current = true;
          } else {
            window.location = redirectURL;
          }
        },
        onFailure: () => { onError(); },
      });
    };
    if (config.googleClientId && !isGoogleInitialized) {
      initializeGoogle();
      setIsGoogleInitialized(true);
    }
  }, [
    Identity,
    config.googleClientId,
    isGoogleInitialized,
    onError,
    redirectURL
  ]);

  useEffect(() => {
    const initializeFacebook = async () => {
      await Identity.initFacebookLogin(null);
    };
    if (config.facebookAppId && !isFacebookInitialized) {
      initializeFacebook();
      setIsFacebookInitialized(true);
    }
  }, [
    Identity,
    config.facebookAppId,
    isFacebookInitialized,
  ]);
```

Reply:
FWIW, this change request still has the the automatic login issue.  It's mainly just to separate the logic of the two social sign ins.  I tested it and it doesn't cause any additional (nor fewer) re-renders.  Just code cleanup.

Reply:
It seems like the logged in logic is doing that redirect, which I think is intended so ignore my comment on being automatically logged in.  The only request I have then is to split that useEffect logic.

Reply:
@nschubach I updated the side effects to Identity, google and facebook effects